packages/listview/src/components/ColumnHeaderFilterText/index.js

```diff
@@ -0,0 +1,103 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import HeaderFilter from './HeaderFilter';
+import FilterOverlay from './FilterOverlay';
+import FilterText from './FilterText';
+import { clearFilters, getHeaderText } from './helpers';
+
+function ColumnHeaderFilterText(props) {
+    const {
+        defaultFilters,
+        onFilter,
+        header,
+        onSort,
+        sortDirection,
+        sortable,
+        headerAlignment,
+    } = props;
+
+    const triggerRef = useRef();
+    const [isOpen, setIsOpen] = useState(false);
+    const [filters, setFilters] = useState(defaultFilters);
+    const [hasFilter, setHasFilter] = useState(defaultFilters.length > 0);
```

Comment:
managing states introduce complexity, then we should try to avoid using states only when strictly necessary and stick to functional, here hasFilter can be computed from filters state